lib/roger/partition/worker.ex

```diff
@@ -48,62 +48,113 @@ defmodule Roger.Partition.Worker do
       channel: channel,
       meta: meta,
       raw_payload: payload}
+    Process.flag(:trap_exit, true)
     {:ok, state, 0}
   end
 
+  def terminate(_reason, state) do
+    if state.worker_task_pid do
+      Process.exit(state.worker_task_pid, :kill)
+    end
+  end
+
+  def handle_info(:handle_job_timeout, %{worker_task_pid: pid, job: job} = state) when is_pid(pid) do
+    Process.exit(pid, :kill)
+    handle_error(job, {:timeout, "Job stopped because of timeout"}, nil, state)
+    {:stop, :normal, state}
+  end
+
   def handle_info(:timeout, state) do
     case Job.decode(state.raw_payload) do
       {:ok, job} ->
         job = %Job{job | started_at: Roger.now}
         if Global.cancelled?(state.partition_id, job.id, :remove) do
```

Comment:
Can this be broken down into smaller functions? It's quite large and somewhat hard to follow